org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -496,6 +492,14 @@ private Map<org.eclipse.jdt.internal.compiler.env.ICompilationUnit, CompilationU
 				}
 			});
 		};
+		// must be 1st thing added to context
+		// also, needs to be added before javac's Log class instance is created
+		context.put(DiagnosticListener.class, diagnosticListener);
+		TolerantJavaCompiler.configureCompilerInstance(context);
```

Comment:
thanks for catching it. I checked and the key point is that we need to register `DiagnosticListener` before initializing the underlying `com.sun.tools.javac.main.JavaCompiler` (aka. `TolerantJavaCompiler`). It's because the JavaCompiler constructor will fetch the the registered `diagnosticListener` to instance the Log. The reorder of other lines in this PR are unnecessary. But since they do no harm, it's ok to merge with this.